src/pip/_internal/operations/build/build_tracker.py

```diff
@@ -51,10 +51,20 @@ def get_build_tracker() -> Generator["BuildTracker", None, None]:
             yield tracker
 
 
+class TrackerId(str):
+    """Uniquely identifying string provided to the build tracker."""
+
+
 class BuildTracker:
+    """Ensure that an sdist cannot request itself as a setup requirement.
+
+    When an sdist is prepared, it identifies its setup requirements in the
+    context of ``BuildTracker#track()``. If a requirement shows up recursively, this
+    raises an exception. This stops fork bombs embedded in malicious packages."""
```

Comment:
```suggestion
    raises an exception.

    This stops fork bombs embedded in malicious packages."""
```

This makes the point that this is about fork bomb protection more obvious. I don't know about others, but for me, the long and complex discussions about exposure to, and prevention of, fork bombs resulting from PEP 518 support still brings back memories. So making it as clear as possible that "hey, don't mess with this unless you know what you're doing" is worthwhile.

Reply:
Yes, absolutely agreed!